// @awrminkhodaei/react-native-horizontal-datepicker/lib/commonjs/components/Selected.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const SelectedDate = ({
  text,
  selectedItemWidth = 170,
  selectedItemTextStyle = styles.selectedDateText,
  selectedItemBackgroundColor = '#16213e',
  itemRadius,
  itemHeight
}) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.selectedDateContainer, {
      width: selectedItemWidth,
      backgroundColor: selectedItemBackgroundColor,
      borderRadius: itemRadius,
      height: itemHeight
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: [selectedItemTextStyle, styles.selectedDateText]
  }, text));
};

const styles = _reactNative.StyleSheet.create({
  selectedDateContainer: {
    width: 170,
    backgroundColor: '#16213e',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 10
  },
  selectedDateText: {
    fontSize: 16,
    color: '#fff'
  }
});

var _default = SelectedDate;
exports.default = _default;
