// @awrminkhodaei/react-native-horizontal-datepicker/lib/commonjs/components/UnSelected.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const UnSelectedDate = ({
  text,
  unselectedItemWidth = 38,
  unselectedItemTextStyle = styles.selectedDateText,
  unselectedItemBackgroundColor = '#fff',
  itemRadius,
  itemHeight
}) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.selectedDateContainer, {
      width: unselectedItemWidth,
      backgroundColor: unselectedItemBackgroundColor,
      borderRadius: itemRadius,
      height: itemHeight
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: [styles.selectedDateText, unselectedItemTextStyle]
  }, text));
};

const styles = _reactNative.StyleSheet.create({
  selectedDateContainer: {
    width: 38,
    backgroundColor: '#fff',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 10
  },
  selectedDateText: {
    fontSize: 16,
    color: '#16213e'
  }
});

var _default = UnSelectedDate;
exports.default = _default;
